test/arrayops.jl

```diff
@@ -2343,3 +2343,18 @@ Base.view(::T25958, args...) = args
     @test t[end,end,1]   == @view(t[end,end,1])   == @views t[end,end,1]
     @test t[end,end,end] == @view(t[end,end,end]) == @views t[end,end,end]
 end
+
+# resize! sets the last element to zero when element size is one byte
+@testset "check that resize! is a no-op when size does not change" for T in (Int, UInt8, Int8)
+    x = Vector{T}(uninitialized, 1)
+    resize!(x, 0)
+    unsafe_store!(pointer(x), 1, 1)
```

Comment:
I don't think this test is valid --- it does not need to be possible to store an element to an empty array. In the future the array's data pointer could be NULL, or tooling could flag this as a memory error, etc.

Reply:
Yeah, it's pretty hacky. I guess we can leave this one out and just rely on the `String`-based test below. I'm not sure whether we can find other cases where this bug would have user-visible consequences we could test.

Reply:
I've removed the test and squashed the commits.